src/ServiceControl.Audit/Infrastructure/RavenDB/MigrateKnownEndpoints.cs

```diff
@@ -0,0 +1,88 @@
+﻿using NServiceBus;
+using NServiceBus.Features;
+using Raven.Client;
+using ServiceControl.Audit.Monitoring;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace ServiceControl.Audit.Infrastructure.RavenDB
+{
+    class MigrateKnownEndpoints : Feature
+    {
+        public MigrateKnownEndpoints()
+        {
+            EnableByDefault();
+        }
+
+        protected override void Setup(FeatureConfigurationContext context)
+        {
+            context.RegisterStartupTask(b => new KnownEndpointMigrator(b.Build<IDocumentStore>()));
+        }
+    }
+
+    class KnownEndpointMigrator : FeatureStartupTask
+    {
+        IDocumentStore store;
+
+        public KnownEndpointMigrator(IDocumentStore store)
+        {
+            this.store = store;
+        }
+
+        protected override async Task OnStart(IMessageSession messageSession)
+        {
+            var knownEndpointsIndex = await store.AsyncDatabaseCommands.GetIndexAsync("EndpointsIndex").ConfigureAwait(false);
+            if (knownEndpointsIndex == null)
+            {
+                return;
+            }
+
+            await WaitForNonStaleIndex(knownEndpointsIndex.Name).ConfigureAwait(false);
+            
+            using (var session = store.OpenAsyncSession())
+            {
+                var endpointsFromIndex = await session.Query<dynamic>(knownEndpointsIndex.Name, true).ToListAsync().ConfigureAwait(false);
```

Comment:
Assumption: The list returned will be small enough to work in memory

Reply:
I think that is a fair assumption considering the complexity of running multiple endpoints as well as our new licensing model. Should we set an explicit limit so that it doesn't crash should there be more though?

Reply:
Paging has been added